src/getCypressElementCoordinates.ts

```diff
@@ -88,3 +88,36 @@ export function getCypressElementCoordinates(
     y: appFrameY + (window.pageYOffset + posY) * appFrameScale,
   };
 }
+
+/**
+ * Returns the coordinates and size of a given Element, relative to the Cypress app <iframe>.
+ */
+function getElementPositionXY(htmlElement: HTMLElement) {
+  const {
+    x: startingX,
+    y: startingY,
+    width,
+    height,
+  }= htmlElement.getBoundingClientRect();
+
+  let finalX = startingX;
+  let finalY = startingY;
+
+  // If the app rendered within the Cypress frame has its own iframes,
+  // we navigate back up the tree of iframes to get the position relative to the Cypress window.
+  let currentWindow: Window | null = htmlElement.ownerDocument.defaultView;
+  while (currentWindow && currentWindow.frameElement && currentWindow !== window) {
+    const { x, y } = currentWindow.frameElement.getBoundingClientRect();
+    finalX += x;
+    finalY += y;
+
+    currentWindow = currentWindow.parent;
+  }
+
+  return {
+    x: finalX,
```

Comment:
Also it is required to calculate the real size of the iframe (if it was scaled). Just like we are doing already for the upper iframe